backend/routes/knowledge_public.py

```
@@ -0,0 +1,329 @@
+        where_clause = {}
+        if content_type:
+            where_clause["content_type"] = content_type
+        if source_filter:
+            where_clause["source"] = {"$contains": source_filter}
+
+        # Get documents using proper encapsulated method
+        try:
+            result_docs = retriever.semantic_searcher.get_documents(
+                where=where_clause if where_clause else None, limit=limit, offset=offset
+            )
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)